packages/gatsby-core-utils/src/fetch-remote-file.ts

```diff
@@ -72,9 +74,63 @@ const ERROR_CODES_TO_RETRY = [
   `ERR_GOT_REQUEST_ERROR`,
 ]
 
+/********************
+ * Queue Management *
+ ********************/
+
+const GATSBY_CONCURRENT_DOWNLOAD = process.env.GATSBY_CONCURRENT_DOWNLOAD
+  ? parseInt(process.env.GATSBY_CONCURRENT_DOWNLOAD, 10) || 0
+  : 200
```

Comment:
This variable will limit it per CPU core not for the whole project. I would lower default to something like 50 instead of 200.

It's probably best to move this to single threaded queue in the future but okay for now.

Reply:
I could move it to https://github.com/sindresorhus/p-queue which has a more simplified interface. It was working great while being part of gatsby-source-contentful.

Reply:
Moved it to 50 per core.